src/components/CheckEditor/FormComponents/HttpCheckRegExValidation.tsx

```diff
@@ -48,44 +68,41 @@ export const HttpCheckRegExValidation = () => {
                           {...rest}
                           aria-label={`Validation Field Name ${userIndex}`}
                           placeholder="Field name"
-                          options={HTTP_REGEX_VALIDATION_OPTIONS}
-                          invalid={
-                            disallowBodyMatching &&
-                            Boolean(errors?.settings?.http?.regexValidations?.[index]?.matchType)
-                          }
+                          options={options}
                           onChange={({ value }) => onChange(value)}
                         />
                       );
                     }}
-                    rules={{
-                      validate: (value) => {
-                        if (disallowBodyMatching) {
-                          if (value?.value === HttpRegexValidationType.Body) {
-                            return 'Cannot validate the body of a HEAD request';
-                          }
-                          return;
-                        }
-                        return;
-                      },
-                    }}
                     name={`${REGEX_FIELD_NAME}.${index}.matchType`}
                   />
                 </div>
                 <div className={styles.validationExpressions}>
                   {isHeaderMatch && (
                     <div className={styles.validationHeaderName}>
-                      <Input
-                        {...register(`${REGEX_FIELD_NAME}.${index}.header`)}
-                        placeholder="Header name"
-                        data-fs-element={`Regex header name ${index}`}
-                      />
+                      <Field
+                        className={styles.field}
+                        invalid={isHttpRegexHeaderError(baseErrorPath) && Boolean(baseErrorPath?.header) ? true : false}
+                        error={isHttpRegexHeaderError(baseErrorPath) && baseErrorPath?.header?.message}
+                      >
+                        <Input
+                          {...register(`${REGEX_FIELD_NAME}.${index}.header`)}
+                          placeholder="Header name"
+                          data-fs-element={`Regex header name ${index}`}
+                        />
+                      </Field>
                     </div>
                   )}
-                  <Input
-                    {...register(`${REGEX_FIELD_NAME}.${index}.expression`)}
-                    placeholder="Regex"
-                    data-fs-element={`Regex expression ${index}`}
-                  />
+                  <Field
+                    className={cx(styles.field, styles.grow)}
+                    invalid={Boolean(baseErrorPath?.expression) ? true : false}
```

Comment:
```suggestion
                    invalid={Boolean(baseErrorPath?.expression)}
```

Reply:
Feel free to disregard though if you think the other option is better for legibility.